pkg/framework/machinesets.go

```diff
@@ -123,7 +124,13 @@ func GetMachineSets(client runtimeclient.Client, selectors ...*metav1.LabelSelec
 		return nil, fmt.Errorf("error querying api for machineSetList object: %w", err)
 	}
 
-	return machineSetList.Items, nil
+	machineSets := []*mapiv1beta1.MachineSet{}
+	for _, ms := range machineSetList.Items {
+		machineSet := ms
```

Comment:
Shouldn't this be a `.DeepCopy()`? This is just assigning the same pointer. Am I missing something here?